<launch>
  <arg name="determination_control_values_param_path" default="$(find-pkg-share determination_control_values)/config/determination_control_values.param.yaml"/>
  <arg name="whill_modelc_param_path" default="$(find-pkg-share ros2_whill)/sample_param.yaml"/>

  <node pkg="determination_control_values" exec="determination_control_values_node" name="determination_control_values" output="screen">
  	<param from="$(var determination_control_values_param_path)"/>
  	<remap from="~/input/joy" to="/whill/states/joy"/>
  	<remap from="~/input/objects" to="/perception/object_recognition/objects"/>
    <remap from="~/input/pointcloud" to="/perception/obstacle_segmentation/pointcloud"/>
    <remap from="~/output/vehicle_cmd" to="/whill/controller/joy"/>

  	<remap from="~/input/whill_odom" to="/whill/odom"/>
    <remap from="~/output/aw_velocity" to="/vehicle/status/velocity_status"/>
    <remap from="~/output/aw_steering" to="/vehicle/status/steering_status"/>
  </node>


  <node pkg="ros2_whill" exec="whill_modelc_controller" name="whill_modelc_controller" output="screen">
    <param from="$(var whill_modelc_param_path)"/>
  </node>
  <node pkg="ros2_whill" exec="whill_modelc_publisher" name="whill_modelc_publisher" output="screen">
    <param from="$(var whill_modelc_param_path)"/>
  </node>

</launch>